crates/bevy_ui/src/render/mod.rs

```diff
@@ -693,10 +712,36 @@ pub fn prepare_uinodes(
     ui_pipeline: Res<UiPipeline>,
     mut image_bind_groups: ResMut<UiImageBindGroups>,
     gpu_images: Res<RenderAssets<Image>>,
-    mut phases: Query<&mut RenderPhase<TransparentUi>>,
     events: Res<SpriteAssetEvents>,
     mut previous_len: Local<usize>,
+    pipeline_cache: Res<PipelineCache>,
+    draw_functions: Res<DrawFunctions<TransparentUi>>,
+    mut render_phases: ParamSet<(
+        Query<&mut RenderPhase<TransparentUi>>,
+        Query<(&ExtractedView, &mut RenderPhase<TransparentUi>)>,
+    )>,
+    mut pipelines: ResMut<SpecializedRenderPipelines<UiPipeline>>,
 ) {
+    let draw_function = draw_functions.read().id::<DrawUi>();
+
+    for (view, mut transparent_phase) in &mut render_phases.p1() {
```

Comment:
This approach goes against the design of the renderer. How come you didn't do the queuing of phase items to phases based on the ranges in queue and then let the phase sort them? This is necessary for flexibility to allow plugins to also queue items to the phase.